internal/tools/aws/client.go

```diff
@@ -2,12 +2,45 @@ package aws
 
 import (
 	"github.com/aws/aws-sdk-go/service/acm"
+	"github.com/aws/aws-sdk-go/service/acm/acmiface"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/aws/aws-sdk-go/service/rds/rdsiface"
 	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/aws/aws-sdk-go/service/route53/route53iface"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
 	"github.com/mattermost/mattermost-cloud/model"
 	log "github.com/sirupsen/logrus"
 )
 
+// AWSClient is a singleton instance of an AWS client.
+var AWSClient *Client
+
+func init() {
+	// Create a singleton instance of an AWS session.
+	sess, err := NewAWSSessionWithLogger(log.WithField("tools-aws", "client"))
+	if err != nil {
+		log.Fatalf("failed to initialize AWS session: %s", err.Error())
```

Comment:
Very good point. Based on our discussion, I changed it to allow multiple clients and also not crash in case of session failure. If a session fails the client still instantiates but obviously any call made to aws will also fail. Still under the session failure scenario, the instance will attempt to get a heath session on every api call, if it succeed, the client instance will construct all clients with the healthy session and cache it.